src/components/dock/dock-button/dock-button.tsx

```diff
@@ -124,14 +124,22 @@ export class DockButton {
                     selected: this.item?.selected,
                 }}
                 onClick={handleClick}
+                aria-live={this.item.badge ? 'polite' : 'off'}
```

Comment:
This is good because we want to let the assistive technologies know when there is an update in here (a notification). However, I don't know how the screen readers work really. :( When I test with Apple's screen readers, I feel everything is so freaking stupid. The reader doesn't even read out loud `aria-described-by`. We should really educate ourselves much more about this and go through the elements.

Reply:
```suggestion
                aria-live={this.item.badge !== undefined ? 'polite' : 'off'}
```

Reply:
I actually have no clue. These things are so poorly explained in their docs. I never know how they work. My assumption is that as soon as there is a notification (which didn't exist before), the user with assistive technologies must realize it too. In case of urgent changes, we would need `role="alert"` or `aria-live="assertive"` [more...](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Alert_Role). 

But since the notification badge is not too serious to scream out loud, they have the `aria-live="polite"`. How screen-readers actually tell the blind user about this little change in the content, I have no idea. 

I even inspected the web-based version of Apple's icloud to see if they have something similar, where there are badges 👇 :
<img width="442" alt="image" src="https://user-images.githubusercontent.com/35954987/220332046-1adceb62-ea07-4a5c-b3c2-c1dde4ebaf64.png">
There was nothing in there.

Then I tried inspecting the web-based version of Microsoft Teams to see how they do it
<img width="338" alt="image" src="https://user-images.githubusercontent.com/35954987/220332487-3d337d4f-4c87-4e22-83c3-56a9ce105780.png">
But their shit didn't even load in the browser anymore and kept showing unknown errors. So I gave up.
<img width="448" alt="image" src="https://user-images.githubusercontent.com/35954987/220332714-e459d641-b4f7-41f1-9b3b-fc73e8c634d4.png">

Reply:
regarding:
`aria-live={this.item.badge !== undefined ? 'polite' : 'off'}`

I still don't know how we will do with `limel-badge`. What if we can just do so in the config of the dock:
```tsx
private dockItems: DockItem[] = [
        {
            id: 'home',
            label: 'Home',
            selected: true,
            icon: '-lime-logo-go-filled',
            badge: '',
        },
```

Is this counted as `undefined`?

Reply:
In case this isn't where you've been reading already, this page is helpful: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-live

Reply:
My point is that if we set `aria-live="off"` while we're not displaying a notification badge, and then `aria-live="polite"` when we show a notification badge, I believe we're only marking it as a "live region" _after_ we have dynamically updated it, so there's nothing. If we use `aria-live` we should have it set regardless of whether we're currently showing a badge or not. We should then set `aria-live="polite"` if we expect that we _might_ show a notification badge.

That is why I suggested that we change the check to `this.item.badge !== undefined` so we would still set `aria-live="polite"` when `this.item.badge === 0` for example.

Reply:
Maybe we don't need to nitpick this too deeply, since we all agree that we're basically just guessing… I already worry about this going down unnecessary rabbit holes… 😬